File: src/p1206/Main.c
#include<stdio.h>
#include<stdlib.h>
long  gcd(long int a,long int b)
{
    long  c;
    
    while(b!=0)
    {
    	c=b;
		b=a%b;
		a=c;
    }
    return a;
}

int main()
{
    long  a,b,n;
    while (scanf("%ld %ld %ld",&a,&b,&n)!=EOF)
    {  
        if (a==0 && b==0 && n==0)  
		{
			printf("Possible!\n");continue;
		}
        if (a==0 && b==0 && n!=0)  
		{
			printf("Impossible!\n");continue;
		}
        long c = gcd(a,b);
        if (n%c == 0)
          printf("Possible!\n");
        else  
          printf("Impossible!\n"); 
        }
    return 0;
    }

File: src/p1206/Main.java
package p1206;

import java.util.Scanner;

/**
 * Copyright(C) 2015,Cao Qingqing , Wuhan University All rights reserved.
 * 
 * This class is submitted for solving the problem 1206 in WOJ
 * 
 * @author : Qingqing Cao, Computer School, Wuhan University
 * 
 */
public class Main {
	static long gcd(long a, long b) {
		long c;

		while (b != 0) {
			c = b;
			b = a % b;
			a = c;
		}
		return a;
	}

	public static void main(String args[]) {

		Scanner input = new Scanner(System.in);

		while (input.hasNext()) {
			int a = input.nextInt();
			int b = input.nextInt();
			int n = input.nextInt();
			if (a == 0 && b == 0 && n == 0) {
				System.out.println("Possible!");
				continue;
			}
			if (a == 0 && b == 0 && n != 0) {
				System.out.println("Impossible!");
				continue;
			}

			long c = gcd(a, b);
			if (n % c == 0)
				System.out.println("Possible!");
			else
				System.out.println("Impossible!");
		}

		input.close();
	}
}
